var/spack/repos/builtin/packages/go/package.py

```diff
@@ -134,8 +138,8 @@ def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
                 path_components.append(d.prefix)
 
         # This *MUST* be first, this is where new code is installed
-        spack_env.set('GOPATH', ':'.join(path_components))
+        env.set('GOPATH', ':'.join(path_components))
```

Comment:
This one should actually be in a separate call to `setup_dependent_build_environment`. `spack_env` corresponds to the build environment used in Spack, `run_env` corresponds to the module environment used when you run the package. Sorry I didn't catch this earlier.

Reply:
Can you please do the modification above? Apart from that this looks ready to be merged.

Reply:
Thanks, I just made the modifications, I should have just added it last week but ran out of time.

Question, I added a function` def generate_path_components(self, dependent_spec):` since `setup_dependent_build_environment` and `setup_dependent_run_environment` both use the same list of dependent paths. Is that okay to do in a Spack package?